### ISSUE TITLE
Report bugs with costant column in Spark DataFrame 

### ISSUE THREAD

**Issue by francoisWeber** [ydataai/pandas-profiling#897]
Hi PandasProfiling Team,
I am testing your `spark-branch` and I encountered the a bug with a Spark DataFrame holding constant columns. Full stack-trace here:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/IPython/core/formatters.py in __call__(self, obj)
    343             method = get_real_method(obj, self.print_method)
    344             if method is not None:
--> 345                 return method()
    346             return None
    347         else:

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/profile_report.py in _repr_html_(self)
    422     def _repr_html_(self) -> None:
    423         """The ipython notebook widgets user interface gets called by the jupyter notebook."""
--> 424         self.to_notebook_iframe()
    425 
    426     def __repr__(self) -> str:

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/profile_report.py in to_notebook_iframe(self)
    402         with warnings.catch_warnings():
    403             warnings.simplefilter("ignore")
--> 404             display(get_notebook_iframe(self.config, self))
    405 
    406     def to_widgets(self) -> None:

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/report/presentation/flavours/widget/notebook.py in get_notebook_iframe(config, profile)
     73         output = get_notebook_iframe_src(config, profile)
     74     elif attribute == IframeAttribute.srcdoc:
---> 75         output = get_notebook_iframe_srcdoc(config, profile)
     76     else:
     77         raise ValueError(

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/report/presentation/flavours/widget/notebook.py in get_notebook_iframe_srcdoc(config, profile)
     27     width = config.notebook.iframe.width
     28     height = config.notebook.iframe.height
---> 29     src = html.escape(profile.to_html())
     30 
     31     iframe = f'<iframe width="{width}" height="{height}" srcdoc="{src}" frameborder="0" allowfullscreen></iframe>'

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/profile_report.py in to_html(self)
    372 
    373         """
--> 374         return self.html
    375 
    376     def to_json(self) -> str:

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/profile_report.py in html(self)
    185     def html(self) -> str:
    186         if self._html is None:
--> 187             self._html = self._render_html()
    188         return self._html
    189 

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/profile_report.py in _render_html(self)
    287         from pandas_profiling.report.presentation.flavours import HTMLReport
    288 
--> 289         report = self.report
    290 
    291         with tqdm(

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/profile_report.py in report(self)
    179     def report(self) -> Root:
    180         if self._report is None:
--> 181             self._report = get_report_structure(self.config, self.description_set)
    182         return self._report
    183 

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/profile_report.py in description_set(self)
    161     def description_set(self) -> Dict[str, Any]:
    162         if self._description_set is None:
--> 163             self._description_set = describe_df(
    164                 self.config,
    165                 self.df,

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/model/describe.py in describe(config, df, summarizer, typeset, sample)
    177         table_stats = table_stats.dict()
    178 
--> 179         alerts = progress(get_alerts, pbar, "Get alerts")(
    180             config, table_stats, series_description, correlations
    181         )

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/utils/progress_bar.py in inner(*args, **kwargs)
      9     def inner(*args, **kwargs) -> Any:
     10         bar.set_postfix_str(message)
---> 11         ret = fn(*args, **kwargs)
     12         bar.update()
     13         return ret

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/model/alerts.py in get_alerts(config, table_stats, series_description, correlations)
    319     alerts = check_table_alerts(table_stats)
    320     for col, description in series_description.items():
--> 321         alerts += check_variable_alerts(config, col, description)
    322     alerts += check_correlation_alerts(config, correlations)
    323     alerts.sort(key=lambda alert: str(alert.alert_type))

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/model/alerts.py in check_variable_alerts(config, col, description)
    287             alerts += categorical_alerts(config, description)
    288         if description["type"] == "Numeric":
--> 289             alerts += numeric_alerts(config, description)
    290 
    291     for idx in range(len(alerts)):

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/model/alerts.py in numeric_alerts(config, summary)
    137 
    138     # Skewness
--> 139     if skewness_alert(summary["skewness"], config.vars.num.skewness_threshold):
    140         alerts.append(
    141             Alert(

~/Code/spark-utils/pysparq/.env/lib/python3.9/site-packages/pandas_profiling/model/alerts.py in skewness_alert(v, threshold)
    330 
    331 def skewness_alert(v: float, threshold: int) -> bool:
--> 332     return not np.isnan(v) and (v < (-1 * threshold) or v > threshold)
    333 
    334 

TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

To reproduce:
```python
df_fake = spark.createDataFrame(data=pd.DataFrame(data={"colA": [1, 1, 1, 1], "colB": [51, 2, 3, 4]}))
a = ProfileReport(df_fake, minimal=True)
a
```

I'm using Python 3.9.9 on a M1 Mac. 

Hope you'll find a way to deal with constant columns.

François

**Comment by sbrugman** [comment]
@chanedwin

**Comment by chanedwin** [comment]
Thanks! This is useful, will investigate

**Comment by chanedwin** [comment]
Hi @francoisWeber,

Sorry been unable to replicate the error with python 3.9.

Could I get your versions for pyspark, python as well as the versions of your installed libraries? This would help replicate the issue. Thanks!

![image](https://user-images.githubusercontent.com/17007874/147627322-3f372554-0967-4365-86af-2cfdafe386e8.png)

**Comment by francoisWeber** [comment]
Hi @chanedwin,
Thank you for digging into this problem. Here is my setup:
- Python 3.9.9 (virtualenv)
- Spark 3.2.0
- libraries versions below (isolated through a virtualenv + piptools/pip-compile to lock the dependencies versions)
```
appnope==0.1.2
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
attrs==21.2.0
backcall==0.2.0
bleach==4.1.0
certifi==2021.10.8
cffi==1.15.0
charset-normalizer==2.0.9
click==8.0.3
cycler==0.11.0
debugpy==1.5.1
decorator==5.1.0
defusedxml==0.7.1
entrypoints==0.3
fonttools==4.28.3
htmlmin==0.1.12
idna==3.3
imagehash==4.2.1
ipykernel==6.6.0
ipython==7.30.1
ipython-genutils==0.2.0
ipywidgets==7.6.5
jedi==0.18.1
jinja2==3.0.3
joblib==1.1.0
jsonschema==4.2.1
jupyter-client==7.1.0
jupyter-core==4.9.1
jupyterlab-pygments==0.1.2
jupyterlab-widgets==1.0.2
kiwisolver==1.3.2
markupsafe==2.0.1
matplotlib==3.5.1
matplotlib-inline==0.1.3
minio==7.1.2
missingno==0.5.0
mistune==0.8.4
multimethod==1.6
nbclient==0.5.9
nbconvert==6.3.0
nbformat==5.1.3
nest-asyncio==1.5.4
networkx==2.6.3
notebook==6.4.6
numpy==1.21.4
packaging==21.3
pandas==1.3.4
pandas-profiling @ https://github.com/pandas-profiling/pandas-profiling/archive/refs/heads/spark-branch.zip
pandocfilters==1.5.0
parso==0.8.3
pexpect==4.8.0
phik==0.12.0
pickleshare==0.7.5
pillow==8.4.0
prometheus-client==0.12.0
prompt-toolkit==3.0.24
ptyprocess==0.7.0
py4j==0.10.9.2
pycparser==2.21
pydantic==1.8.2
pygments==2.10.0
pyhdfs==0.3.1
pyparsing==3.0.6
pyrsistent==0.18.0
pyspark==3.2.0
python-dateutil==2.8.2
python-dotenv==0.19.2
pytz==2021.3
pywavelets==1.2.0
pyyaml==6.0
pyzmq==22.3.0
requests==2.26.0
scipy==1.7.3
seaborn==0.11.2
send2trash==1.8.0
simplejson==3.17.6
six==1.16.0
smart-open==5.2.1
tangled-up-in-unicode==0.2.0
terminado==0.12.1
testpath==0.5.0
tornado==6.1
tqdm==4.62.3
traitlets==5.1.1
typing-extensions==4.0.1
urllib3==1.26.7
visions[type_image_path]==0.7.4
wcwidth==0.2.5
webencodings==0.5.1
widgetsnbextension==3.5.2
```

**Comment by chanedwin** [comment]
Thanks again! This helps tons

**Comment by chanedwin** [comment]
Issue identified and fixed in #903.
Main issue was how spark functions skewness and kurtosis return None instead of np.nan in newer versions of spark for constant columns, resulting in downstream errors. 

@sbrugman

**Comment by chanedwin** [comment]
merged to spark-branch, please try! @francoisWeber . Will close if all good

**Comment by francoisWeber** [comment]
Hi @chanedwin . It worked on a minimalist Virtualenv but my main project is not OK yet. Will try to run the update on this latter too ! I will keep you updated

**Comment by francoisWeber** [comment]
OK it works fine now in my whole project 👍 Thanks a lot :)

**Comment by chanedwin** [comment]
awesome!

**Comment by fabclmnt** [comment]
Closing the issue as it was solved by #903.
